content/articles/how-to-implement-object-oriented-programming-in-matlab/index.md

```diff
@@ -0,0 +1,178 @@
+return
+end
+function S = Salary(objS, noOfHours)
+S = (objS.baseSalary*noOfHours)*objS.commission;
+end
+end
+end
+```
+The `objE` is the output of the subclass constructor and the read superclass name and any other argument required by the argument superclass.
+In Matlab, you can check the final properties of the class by the command properties and the class name. 
+```Matlab
+properties Engineers
+```
+This property has all the public properties of the superclass employees.
+
+![properties engineer](/engineering-education/how-to-implement-object-oriented-programming/oop_two.png)
+
+### Polymorphism
+It is a technique in object-oriented programming that allows you to eliminate long lines of codes of `if`, `else`, or `switch case` statements. There are two types of polymorphism;
+To implement this in Matlab, we look at the example we used in the cMatlabngineers implemented in the `function salary.`
+```Matlab
+% function for salary for an engineer
+function y = Salary(objE, noOfHours) 
+y = objE.baseSalary*noOfHours;
+return
+end
+```
+
+```matlab
+% function for salary for sales
+function S = Salary(objS, noOfHours)
+S = (objS.baseSalary*noOfHours)*objS.commission;
+end
+```
+In the above code, we have a function salary. Both the engineer and sales class have the same input argument, but they calculate the salary using different mathematical operations. Here we initiate our object of each class `objE` and `objS`. We use the name and the base salary from the superclass and provide new values for their properties. For example, we had our employees 'neha'. We must specify the products and sales of neha. 
+We first specify our objects i.e `objE` and `objS`. To define `objE`,
+```matlab
+objE = Engineer(obj.name, obj.baseSalary, "simulink")
+```
+![object engineer](/engineering-education/how-to-implement-object-oriented-programming/oop_Three.png)
+The parameters of our object are the name, salary, and the product, which is Simulink.
+```Matlab
+objS = Sales(obj.name, obj.baseSalary, "NA", 500)
+```
+Here the parameters are name, salary, region, which in this case is not applicable(NA), and commission which is 500
+![object sales](/engineering-education/how-to-implement-object-oriented-programming/oop_four.png)
+
+To see the impact of these, we will execute the following command in the command window. 
+```matlab
+SE = Salary(objE, 10)  %Calculates the salary
+
+SS =  Salary(objS,10)   %calculates the salary from sales
+```
+The value 10 is the number of hours. When we run the codes separately, we get the salary and the sales.
+
+![image showing results](/engineering-education/how-to-implement-object-oriented-programming/oop_five.png)
+The calculated salary is different as per the mathematical operations.
+
+### Abstraction
+It can be said to be an extension of encapsulation. To implement it in our example, we can only define the function named `salary` in the base class employed, and it's derived will have the function implementation. 
```

Comment:
Hey seems you've used "It" a lot. Kindly rewrite this in every instance of the OOP introduction to improve readability.